tests/test_distribution.py

```diff
@@ -0,0 +1,216 @@
+        dist = Distribution.get_distribution(self.ctx)
+        self.assertEqual(dist.name, "unnamed_dist_1")
+
+    @mock.patch("pythonforandroid.util.chdir")
+    @mock.patch("pythonforandroid.distribution.open", create=True)
+    def test_save_info(self, mock_open_dist_info, mock_chdir):
+        self.setUp_distribution_with_bootstrap(
+            Bootstrap().get_bootstrap("sdl2", self.ctx)
+        )
+        self.ctx.hostpython = "/some/fake/hostpython3"
+        self.ctx.python_recipe = Recipe.get_recipe("python3", self.ctx)
+        self.ctx.python_modules = ["requests"]
+        mock_open_dist_info.side_effect = [
+            mock.mock_open(read_data=json.dumps(dist_info_data)).return_value
+        ]
+        self.ctx.bootstrap.distribution.save_info("/fake_dir")
+        mock_open_dist_info.assert_called_once_with("dist_info.json", "w")
+        mock_open_dist_info.reset_mock()
+
+    @mock.patch("pythonforandroid.distribution.open", create=True)
+    @mock.patch("pythonforandroid.distribution.exists")
+    @mock.patch("pythonforandroid.distribution.glob.glob")
+    def test_get_distributions(
+        self, mock_glob, mock_exists, mock_open_dist_info
+    ):
+        self.setUp_distribution_with_bootstrap(
+            Bootstrap().get_bootstrap("sdl2", self.ctx)
+        )
+        mock_glob.return_value = ["sdl2-python3"]
+        mock_open_dist_info.side_effect = [
+            mock.mock_open(read_data=json.dumps(dist_info_data)).return_value
+        ]
+
+        dists = self.ctx.bootstrap.distribution.get_distributions(self.ctx)
+        self.assertIsInstance(dists, list)
+        self.assertEqual(len(dists), 1)
+        self.assertIsInstance(dists[0], Distribution)
+        self.assertEqual(dists[0].name, "sdl2-python3")
+        self.assertEqual(dists[0].ndk_api, 21)
+        self.assertEqual(
+            dists[0].recipes,
+            ["hostpython3", "python3", "sdl2", "kivy", "requests"],
+        )
+        mock_open_dist_info.assert_called_with("sdl2-python3/dist_info.json")
+        mock_open_dist_info.reset_mock()
+
+    @mock.patch("pythonforandroid.distribution.open", create=True)
+    @mock.patch("pythonforandroid.distribution.exists")
+    @mock.patch("pythonforandroid.distribution.glob.glob")
+    def test_get_distributions_error_ndk_api(
+        self, mock_glob, mock_exists, mock_open_dist_info
+    ):
+        dist_info_data_no_ndk_api = dist_info_data.copy()
+        dist_info_data_no_ndk_api.pop("ndk_api")
+        self.setUp_distribution_with_bootstrap(
+            Bootstrap().get_bootstrap("sdl2", self.ctx)
+        )
+        mock_glob.return_value = ["sdl2-python3"]
+        mock_open_dist_info.side_effect = [
+            mock.mock_open(
+                read_data=json.dumps(dist_info_data_no_ndk_api)
+            ).return_value
+        ]
+
+        dists = self.ctx.bootstrap.distribution.get_distributions(self.ctx)
+        self.assertEqual(dists[0].ndk_api, None)
```

Comment:
just a random idea, we could also have mocked `warning` and asserted its call

Reply:
You are right we could do that, but I more concerned to mock the stuff that could slow down our tests (mostly file operations: copy, paste, pack, unpack, read files...)

Reply:
yes I just throw that idea, because I may have mocked it, just to show we're throwing a warning in this specific case. But you're covering the case anyway asserting ndk_api is None. By the way if you like self.assert you also have `self.assertIsNone()`.